openstack_virtual_baremetal/tests/test_deploy.py

```diff
@@ -192,11 +192,11 @@ def test_generate_with_role(self, mock_safe_dump, mock_bed):
 # _process_role test data
 role_base_data = {
     'parameter_defaults': {
-        'overcloud_storage_mgmt_net': 'storage_mgmt-foo',
-        'overcloud_internal_net': 'internal-foo',
-        'overcloud_storage_net': 'storage-foo',
+        'overcloud_storage_mgmt_net': 'storage_mgmt',
```

Comment:
Unit tests was failing:
`git checkout c1364026fa3069072e8ce70261a867e84578d641`
`git checkout -b test-branch`
`git checkout ef30ff2e56d86e3e951dec1fb28269986cf27675 openstack_virtual_baremetal/tests/test_deploy.py `
`tox -e py27`

> FAIL: tests.test_deploy.TestDeploy.test_process_role
> tags: worker-1
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/hjensas/code/ovb-routed/openstack-virtual-baremetal/.tox/py27/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
>     return func(*args, **keywargs)
>   File "/home/hjensas/code/ovb-routed/openstack-virtual-baremetal/openstack_virtual_baremetal/tests/test_deploy.py", line 402, in test_process_role
>     output['parameter_defaults']['overcloud_tenant_net'])
>   File "/home/hjensas/code/ovb-routed/openstack-virtual-baremetal/.tox/py27/lib/python2.7/site-packages/testtools/testcase.py", line 411, in assertEqual
>     self.assertThat(observed, matcher, message)
>   File "/home/hjensas/code/ovb-routed/openstack-virtual-baremetal/.tox/py27/lib/python2.7/site-packages/testtools/testcase.py", line 498, in assertThat
>     raise mismatch_error
> testtools.matchers._impl.MismatchError: 'tenant-foo' != 'tenant-foo-foo'


Maby this is happening because of the change I made to openstack_virtual_baremetal/deploy.py to fix the key error?

Reply:
So instead of the chenges done to openstack_virtual_baremetal/deploy.py, the key error can be fixed by adding to role_base_data = {'parameter_defaults': [...]} as in the patch below.

However I still get the `testtools.matchers._impl.MismatchError: 'tenant-foo' != 'tenant-foo-foo'`
error.

```
diff --git a/openstack_virtual_baremetal/tests/test_deploy.py b/openstack_virtual_baremetal/tests/test_deploy.py
index 07fb293..67a87a0 100755
--- a/openstack_virtual_baremetal/tests/test_deploy.py
+++ b/openstack_virtual_baremetal/tests/test_deploy.py
@@ -196,7 +196,9 @@ role_base_data = {
         'overcloud_internal_net': 'internal-foo',
         'overcloud_storage_net': 'storage-foo',
         'role': 'control',
-        'overcloud_tenant_net': 'tenant-foo'
+        'overcloud_tenant_net': 'tenant-foo',
+        'public_net': 'public-foo',
+        'private_net': 'private',
     },
     'parameters': {
         'os_user': 'admin',
```

Reply:
Okay, that's a legitimate bug. It shouldn't be adding the id when it is inheriting the network names, only when the network name is overridden in the role file. That code should probably check for whether the id is already present before appending it.

However, as that was an existing bug it shouldn't block this. I'll clean that up after we get this PR merged.